framework/src/application/node/synchronizer/block_synchronization_mechanism.ts

```diff
@@ -571,6 +571,7 @@ export class BlockSynchronizationMechanism extends BaseSynchronizer {
 			id: Buffer.alloc(0),
 			height: selectedPeers[randomPeerIndex].height,
 			version: selectedPeers[randomPeerIndex].blockVersion,
+			previousBlockID: Buffer.alloc(0),
```

Comment:
if we always send empty `previousBlockID`. why not ignore this if its not required?

Reply:
It is comparing previousBlockID in the fork choice rule to decide wether to sync or not.
Previously it was string, so comparing with `undefined` was fine, but now it should be at least Buffer to use `buf.equal`